junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
> Related to the actual fix, what I found was that in a test where k3po rule is around the gateway rule, if there was an exception from the gateway statement, then notifyException was called twice, once for the actual exception and a second time when an exception was thrown when k3po was trying to clean up.

Awesome find, I think we can fix the issue by simplifying the code, rather than adding to it for work arounds.